usvm-core/src/main/kotlin/org/usvm/constraints/ULogicalConstraints.kt

```diff
@@ -32,8 +36,8 @@ class ULogicalConstraints private constructor(
     val isContradicting: Boolean
         get() = constraints.any(UBoolExpr::isFalse)
 
-    fun contradiction(ctx: UContext<*>) {
-        constraints = persistentHashSetOf(ctx.falseExpr)
+    fun contradiction(ctx: UContext<*>, ownership: MutabilityOwnership) {
+        constraints = persistentHashSetOf<KExpr<UBoolSort>>().add(ctx.falseExpr, ownership)
```

Comment:
Please, use `UBoolExpr` here. In the USVM we have an invariant, that `KExpr` is an expression, which is translated into SMT solver related representation.